src/features/newsFeed/useNewsFeedStore.tsx

```diff
@@ -0,0 +1,115 @@
+import axios from 'axios';
+import { create } from 'zustand';
+
+import { listNewsFeed } from '@/api/NewsFeedApi';
+import { NewsItemT } from '@/types/NewsFeedTypes';
+
+export const NEWS_FEED_PAGE_SIZE = 10;
+
+type ListNewsFeedArgsT = {
+    firstPage: boolean;
+    timestamp_before?: string;
+};
+
+type NewsFeedStateT = {
+    news: NewsItemT[];
+    newsFeedCurrentPage: number;
+    newsCount: number;
+    unseenNewsCount: number;
+
+    fetchingNews: boolean;
+    fetchingUnseenNewsCount: boolean;
+
+    fetchNews: ({ firstPage, timestamp_before }: ListNewsFeedArgsT) => void;
+    fetchUnseenNewsCount: (timestamp_after: string) => void;
+};
+
+let fetchNewsController: AbortController | undefined;
+let fetchUnseenNewsCountController: AbortController | undefined;
+
+const useNewsFeedStore = create<NewsFeedStateT>((set) => ({
+    news: [],
+    newsFeedCurrentPage: 1,
+    newsCount: 0,
+    unseenNewsCount: 0,
+    fetchingNews: false,
+    fetchingUnseenNewsCount: false,
+    fetchNews: async ({ firstPage, timestamp_before }: ListNewsFeedArgsT) => {
+        // abort previous call
+        if (fetchNewsController) {
+            fetchNewsController.abort();
+        }
+
+        fetchNewsController = new AbortController();
+        set((state) => ({
+            fetchingNews: true,
+            news: firstPage ? [] : state.news,
+        }));
+        try {
+            const page = firstPage
+                ? 1
+                : useNewsFeedStore.getState().newsFeedCurrentPage;
```

Comment:
Instead of calling getState, you can modify the function on line 30 to these args: `(set, get)` and then just use the method get().